lib/java/com/google/android/material/imageview/ShapeableImageView.java

```diff
@@ -219,13 +216,17 @@ public void setContentPadding(
     startContentPadding = UNDEFINED_PADDING;
     endContentPadding = UNDEFINED_PADDING;
 
-    // Super padding is equal to background padding + content padding. Adjust the content padding
-    //  portion of the super padding here:
-    super.setPadding(
-        super.getPaddingLeft() - leftContentPadding + left,
-        super.getPaddingTop() - topContentPadding + top,
-        super.getPaddingRight() - rightContentPadding + right,
-        super.getPaddingBottom() - bottomContentPadding + bottom);
+    // If onMeasure hasn't adjusted padding yet, wait for it to be set to avoid double-applying the
+    //  content padding in onMeasure:
+    if (hasAdjustedPaddingAfterLayoutDirectionResolved) {
```

Comment:
Will this affect API >= 19?

Reply:
Yes, `hasAdjustedPaddingAfterLayoutDirectionResolved` doesn't get set until after `onMeasure` runs after `isLayoutDirectionResolved` (which is an API 19+ method) becomes true. (So really it affects 19+ more than it affects pre-19.)

Reply:
Got it. Don't you need to call setContentPadding() again when hasAdjustedPaddingAfterLayoutDirectionResolved is set to true? So the previous set content paddings can be applied?

Reply:
No. `setContentPadding` does two things:
1. It calls `super.setPadding` according to this class's internal padding contract. This is skipped if this flag is not set, but `super.setPadding` is also called immediately after the flag is set, so this call is covered either way.
2. It sets the private `contentPadding` variables, regardless of whether that flag is set.

The private `contentPadding` variables are only used in the `getContentPadding` and `getPadding` getters. `updateShapeMask` uses the `getPadding` getters to set the shape size, so now I wonder if I need to force a call to `updateShapeMask` immediately after that flag is set too...

Reply:
That's a good catch. I think onSizeChanged() will be called after setting paddings, right? That should be fine I guess.

Reply:
I think `onSizeChanged` is not called in that case actually. I'll need to try to force an error this way to be sure, and add a related test if possible